ui/app/store/actions.js

```diff
@@ -1956,10 +1956,12 @@ function addToAddressBook (recipient, nickname = '', memo = '') {
 
   return (dispatch, getState) => {
     const chainId = getState().metamask.network
-    const set = background.setAddressBook(checksumAddress(recipient), nickname, chainId, memo)
-    if (!set) {
-      return dispatch(displayWarning('Address book failed to update'))
-    }
+    background.setAddressBook(checksumAddress(recipient), nickname, chainId, memo, (_, set) => {
```

Comment:
It looks like the error is being ignored. That should probably be re-thrown, shouldn't it? Unless there's some quirk of dnode involved here that I'm not aware of.

Or instead you could switch this to use `pify`, and await the result. That should throw on error. In general I think we should prefer async/await even just to avoid cases where errors could be swallowed accidentally.

Reply:
Addressed in https://github.com/MetaMask/metamask-extension/pull/6978/commits/967f7f25ee399e37e0b862af759bab6e393b271e